src/config-serializer/config-items/macro-action/EditableMacroAction.ts

```diff
@@ -0,0 +1,198 @@
+
+import {KeyAction, keyActionType} from '../key-action/KeyAction';
+import {KeystrokeAction} from '../key-action/KeystrokeAction';
+import {DelayMacroAction} from './DelayMacroAction';
+import {KeyMacroAction} from './KeyMacroAction';
+import {MacroAction, MacroSubAction, macroActionType} from './MacroAction';
+import {MouseButtonMacroAction} from './MouseButtonMacroAction';
+import {MoveMouseMacroAction} from './MoveMouseMacroAction';
+import {ScrollMouseMacroAction} from './ScrollMouseMacroAction';
+import {TextMacroAction} from './TextMacroAction';
+
+const macroActionTypeValues = Object.keys(macroActionType).map(key => macroActionType[key]);
+
+interface JsObjectEditableMacroAction {
+    macroActionType: string;
+    action?: string;
+    scancode?: number;
+    modifierMask?: number;
+    mouseButtonsMask?: number;
+    x?: number;
+    y?: number;
+    delay?: number;
+    text?: string;
+}
+
+export class EditableMacroAction {
+    macroActionType: string;
+    action: MacroSubAction;
+    // Key macro action properties
+    scancode: number;
+    modifierMask: number;
+    // Mouse macro action properties
+    mouseButtonsMask: number;
+    moveX: number;
+    moveY: number;
+    scrollX: number;
+    scrollY: number;
+    // Delay macro action properties
+    delay: number;
+    // Text macro action properties
+    text: string;
+
+    assertMacroActionType(jsObject: any) {
```

Comment:
Is `any` replacable with `JsObjectEditableMacroAction` here ? If yes, then replace it!